editor/script/camera_controller.gd

```diff
@@ -0,0 +1,26 @@
+extends Node3D
+
+@export var base_cam_speed: float = 10.0
+@export var base_cam_rot_speed: float = 2.0
+
+func _process(delta):
+	camera_movement(delta)
+	camera_rotate(delta)
+
+#Handle input for translation/movement
+func camera_movement(delta) -> void:
+	var direction = Vector3(
+		Input.get_action_strength("camera_right") - Input.get_action_strength("camera_left"),
+		Input.get_action_strength("camera_up") - Input.get_action_strength("camera_down"),
+		Input.get_action_strength("camera_backwards") - Input.get_action_strength("camera_forward"),
+	)
+
+	if(direction != Vector3.ZERO):
+		direction = direction.normalized()
+		var finalSpeed = base_cam_speed * (Input.get_action_strength("camera_boost") +1)
+		translate(direction * finalSpeed * delta)
+
+#Handle input for rotation
+func camera_rotate(delta) -> void:
+	var rot_input = Input.get_action_strength("camera_rot_left") - Input.get_action_strength("camera_rot_right")
+	rotate_y(rot_input * base_cam_rot_speed * delta)
```

Comment:
It took me some time to figure out exactly which buttons rotated the camera, so we should consider adding a corner text with the button mappings later on.

Reply:
Yea could do it in static text for now but I might just create an UI/options menu branch and do it on that to view controls.